macaroons/store.go

```diff
@@ -118,6 +122,77 @@ func (r *RootKeyStorage) CreateUnlock(password *[]byte) error {
 	})
 }
 
+// ChangeEncKey updates the encryption key used to encrypt/decrypt the root keys
+// stored.
+func (r *RootKeyStorage) ChangeEncKey(old, new []byte) error {
+	return r.Update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(rootKeyBucketName)
+		dbKey := bucket.Get(encryptedKeyID)
+
+		// First, we'll make sure the macaroon store has been created.
+		if len(dbKey) == 0 {
+			return ErrStoreNotCreated
+		}
+
+		// If it is, we'll verify that it is correct and create a new
+		// encryption key with the new password.
+		oldEncKey := &snacl.SecretKey{}
+		err := oldEncKey.Unmarshal(dbKey)
+		if err != nil {
+			return err
+		}
+
+		err = oldEncKey.DeriveKey(&old)
+		if err != nil {
+			return err
+		}
+
```

Comment:
after this method returns, we want to clear this secret key from memory using `defer oldEncKey.Zero()`

Reply:
Fixed.